### PR TITLE
[x11] Fix two stacking issues

### LINKED ISSUE #4355: `plank` window not kept above
### The issue:

Following discussion at #4049. Would be interesting to check if related to #1145. Going to try  `keep_above` with `tint2`, although I think this is a `qtile` issue. Also gonna try my hand at a `bisect`.

### Required:

- [X] I have searched past issues to see if this bug has already been reported.

### PR DESCRIPTION
**Pull request by elParaguayo**
1) window.bring_to_front ignored stacking rules. This has been amended so that stacking rules are respected.
2) Static windows were ignored by stacking rules. This has also been amended to ensure that these windows are included.

Fixes #4355

@ervinpopescu - would be great if you could test this.

### CODE REVIEW

**Comment by ervinpopescu** [comment]
I forgot to mention `plank` was `Static`, I'm glad you saw it too. Gonna test this now.

**Comment by ervinpopescu** [comment]
This works as expected, thanks @elParaguayo! I can't figure out why the tests are failing, so I can't help you with that. Looking forward to this getting merged :)

**Comment by elParaguayo** [comment]
Nice.  Thanks for confirming. 

I'll look at tests after work today. Really don't understand why one of them is failing. Probably means I've done something stupid!

**Comment by ervinpopescu** [comment]
Q: Why is the `bring_to_front` behavior changed from the original PR? I thought that you wanted it not to obey the stacking rules.

**Comment by elParaguayo** [comment]
Because that would cause plank to sit behind any window that had been clicked on. 

We could still have a command that puts a window on the top of the stack but I don't think it's compatible with this particular issue,  unless you have an idea...

**Comment by ervinpopescu** [comment]
Well don't we already have `move_to_top`? Sorry if I'm being a dumbo 😭 I didn't have a chance to look into this more.

**Comment by jwijenbergh** [comment]
great jobs on the docs, imo LGTM once we fix test with the other pr?

**Comment by elParaguayo** [comment]
I'll rebase and tidy up history a bit.

**Comment by elParaguayo** [comment]
> Well don't we already have `move_to_top`? Sorry if I'm being a dumbo sob I didn't have a chance to look into this more.

So... I've kept `bring_to_front` behaviour as it it (i.e. ignoring layer grouping rules). Try changing `bring_front_click` to `"floating_only"` in your config. That should stop tiled windows coming above your dock.

**Comment by ervinpopescu** [comment]
> So... I've kept `bring_to_front` behaviour as it it (i.e. ignoring layer grouping rules). Try changing `bring_front_click` to `"floating_only"` in your config. That should stop tiled windows coming above your dock.

Okay, thanks!

File: libqtile/backend/x11/window.py [hunk 1]
```diff
@@ -1615,6 +1622,10 @@ def bring_to_front(self):
         if self.get_wm_type() != "desktop":
             self.window.configure(stackmode=xcffib.xproto.StackMode.Above)
 
+    # @property
+    # def fullscreen(self):
+    #     return False
+
```
Comment (elParaguayo) [inline libqtile/backend/x11/window.py:1628]: Note to self: delete these lines!

**Pull request merged**